draft-ietf-tls-wkech.xml

```diff
@@ -345,120 +342,147 @@ Client <-->|                    | |                    | facing
+      <t>
+	  regeninterval: the number of seconds between key generation
+	  actions at the origin, i.e. a replacement ECHConfigList may be
+	  generated this often.
+      </t>
+    </list>
     </t>
 
-    <t>
-        This arrangement provides the following important properties:
-        <list style="symbols">
-            <t>
-                Origins can indicate that different ECHConfigs are used on
-                different ports.
-            </t>
-            <t>
-                Origins can indicate that multiple CDNs are in use, each with
-                its own ECHConfig.
-            </t>
-            <t>
-                Origins that simply alias to a single target can indicate this
-                without copying the ECHConfig and other parameters, which
-                can interfere with key rotation and other maintenance.
-            </t>
-            <t>
-                "port" and "target" are generally sufficient to uniquely identify
-                a ServiceMode record, so zone factories can use the endpoint list
-                to add ECH to pre-existing ServiceMode records that may have other
-                SvcParams.
-            </t>
-        </list>
+    <t>The following keys are defined for ServiceMode entries:
+    <list style="symbols">
+      <t>
+	  target: The value is a string containing a fully qualified
+	  domain name, corresponding to the HTTPS record's TargetName.
+	  The default value is ".".  All labels in the name MUST contain
+	  only octets from the ABNF set ALPHA / DIGIT / "-" / "_".
+      </t>
+      <t>
+	  priority: The value is a positive integer corresponding to the
+	  SvcPriority.  If omitted, the ZF MAY infer
+	  numerically increasing SvcPriority from the order of the
+	  endpoints array.
+      </t>
+      <t>
+	  params: A JSON Dictionary representing the SVCB SvcParams.  Each
+	  key in the dictionary is a string containing a registered
+	  SvcParamKey name (e.g., "ipv6hint") or a SvcParamKey in generic
+	  form (e.g., "key65528").
+	<list>
+	  <t>
+	      For single-valued SvcParams (e.g., "ech"), the value is a JSON
+	      String.  A JSON String is a sequence of Unicode codepoints,
+	      while a SvcParam's presentation value is a sequence of octets,
+	      so each value octet is stored as a single Unicode codepoint
+	      <xref target="ISOMORPHIC-DECODE"/>.  In almost all cases, this
+	      is equivalent to the ordinary string representation of the
+	      presentation value.
+	  </t>
+	  <t>
+	      For list-valued SvcParams (e.g., "alpn"), the value is a JSON Array
+	      of Strings.  Each String represents an octet sequence, as in the
+	      single-value case.
+	  </t>
+	</list>
+      </t>
+    </list>
+  </t>
+
+    <t>The following key is defined for AliasMode entries.
+
+    <list style="symbols">
+      <t>
+	  alias: The value MUST be a DNS name that could be used as the
+	  TargetName of an HTTPS resource record.  This indicates that the
```

Comment:
I think this name should have the same character restrictions as `target`, whatever those may be.  (RFC 9460 imposes no restrictions on either, but there is a [related interesting wrinkle](https://datatracker.ietf.org/doc/html/rfc9460#section-10.3).)

Reply:
Maybe "same restrictions as `target` but see section 10 of 9460" ?

Reply:
That section applies equally to `target` and `alias`.  I probably just wouldn't mention it in this document.

Reply:
okay, so removed the charset section in that "target" entry up at line 403.